core/src/test/java/org/apache/spark/util/collection/TestTimSort.java

```diff
@@ -0,0 +1,111 @@
+package org.apache.spark.util.collection;
+
+import java.util.*;
+
+/**
+ * This codes generates a int array which fails the standard TimSort, Borrowed from
+ * the reporter of this bug.
+ *
+ * http://www.envisage-project.eu/timsort-specification-and-verification/
```

Comment:
we should also link to the github repository with an explicit license https://github.com/abstools/java-timsort-bug